File: samples/macOS/java/TableManagerSamples/CreateOTO/src/ResponseListener.java
package createoto;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.Semaphore;
import java.util.concurrent.TimeUnit;

import com.fxcore2.*;

public class ResponseListener implements IO2GResponseListener {
    private List<String> mRequestIDs;
    private Semaphore mSemaphore;

    // ctor
    public ResponseListener() {
        mRequestIDs = new ArrayList<String>();
        mSemaphore = new Semaphore(0);
    }

    public void setRequestIDs(List<String> requestIDs) {
        mRequestIDs.clear();
        for (String sOrderID : requestIDs) {
            mRequestIDs.add(sOrderID);
        }
    }

    public boolean waitEvents() throws Exception {
        return mSemaphore.tryAcquire(30, TimeUnit.SECONDS);
    }

    public void stopWaiting() {
        mSemaphore.release();
    }

    // Implementation of IO2GResponseListener interface public method onRequestCompleted
    public void onRequestCompleted(String sRequestID, O2GResponse response) {
    }

    // Implementation of IO2GResponseListener interface public method onRequestFailed
    public void onRequestFailed(String sRequestID, String sError) {
        if (mRequestIDs.contains(sRequestID)) {
            System.out.println("Request failed: " + sError);
            stopWaiting();
        }
    }

    // Implementation of IO2GResponseListener interface public method onTablesUpdates
    public void onTablesUpdates(O2GResponse response) {
    }
}

File: samples/Linux/netcore_for_Linux_x64_only/netcore1.0/TableManagerSamples/CreateOTOCO/src/ResponseListener.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using fxcore2;

namespace CreateOTO
{
    class ResponseListener : IO2GResponseListener
    {
        private List<string> mRequestIDs;
        private EventWaitHandle mSyncResponseEvent = null;

        /// <summary>
        /// ctor
        /// </summary>
        public ResponseListener()
        {
            mRequestIDs = new List<string>();
            mSyncResponseEvent = new EventWaitHandle(false, EventResetMode.AutoReset);
        }

        public void SetRequestIDs(List<string> requestIDs)
        {
            mRequestIDs.Clear();
            foreach (string sOrderID in requestIDs)
            {
                mRequestIDs.Add(sOrderID);
            }
        }

        public bool WaitEvents()
        {
            return mSyncResponseEvent.WaitOne(30000);
        }

        public void StopWaiting()
        {
            mSyncResponseEvent.Set();
        }

        #region IO2GResponseListener Members

        // Implementation of IO2GResponseListener interface public method onRequestCompleted
        public void onRequestCompleted(string sRequestID, O2GResponse response)
        {
        }

        // Implementation of IO2GResponseListener interface public method onRequestFailed
        public void onRequestFailed(string sRequestID, string sError)
        {
            if (mRequestIDs.Contains(sRequestID))
            {
                Console.WriteLine("Request failed: " + sError);
                StopWaiting();
            }
        }

        // Implementation of IO2GResponseListener interface public method onTablesUpdates
        public void onTablesUpdates(O2GResponse response)
        {
        }

        #endregion


    }
}

